packages/aft/lib/src/commands/generate/generate_workflows_command.dart

```diff
@@ -128,6 +130,93 @@ jobs:
         }
       }
       workflowFile.writeAsStringSync(workflowContents.toString());
+
+      await generateAndroidUnitTestWorkflow(
+        package: package,
+        repoRelativePath: repoRelativePath,
+      );
+    }
+  }
+
+  /// If a package has Android unit tests, generate a separate workflow for them.
+  Future<void> generateAndroidUnitTestWorkflow({
+    required PackageInfo package,
+    required String repoRelativePath,
+  }) async {
+    const androidWorkflow = 'flutter_android.yaml';
+
+    final appFacingAndroidTestDir =
+        Directory(p.join(package.path, 'android', 'src', 'test'));
+    final platformAndroidPackageTestDir = Directory(
+      p.join('${package.path}_android', 'android', 'src', 'test'),
+    ); // federated _android package
+    final androidExampleDir = Directory(
+      p.join(package.path, 'example', 'android'),
+    );
+    final appFacingPackageAndroidTestsDirExists =
+        appFacingAndroidTestDir.existsSync();
+    final platformPackageAndroidTestDirExists =
+        platformAndroidPackageTestDir.existsSync();
+    final androidExampleDirExists = androidExampleDir.existsSync();
+    final hasAndroidTests = androidExampleDirExists &&
+        (appFacingPackageAndroidTestsDirExists ||
+            platformPackageAndroidTestDirExists);
+
+    if (package.flavor != PackageFlavor.flutter || !hasAndroidTests) {
+      return;
     }
+
+    final androidWorkflowFilepath = p.join(
+      rootDir.path,
+      '.github',
+      'workflows',
+      '${package.name}_android.yaml',
```

Comment:
Can we rename these to `{package.name}.android.yaml` to distinguish from a package named `{package.name}_android`?